React/Fabric/Mounting/ComponentViews/TextInput/RCTTextInputComponentView.mm

```diff
@@ -617,6 +621,26 @@ - (void)_setMultiline:(BOOL)multiline
   RCTCopyBackedTextInput(_backedTextInputView, backedTextInputView);
   _backedTextInputView = backedTextInputView;
   [self addSubview:_backedTextInputView];
+
+  auto const &currentTextInputProps = *std::static_pointer_cast<TextInputProps const>(_props);
+  [self _setShowSoftInputOnFocus:currentTextInputProps.traits.showSoftInputOnFocus];
```

Comment:
why do we couple this logic with multiline? regardless, the initial props should go through the updateProps:oldProps pipeline.

Reply:
Sorry for the late reply, I must've missed the notification 😅. It's there because setting `multiline` causes the creation of a new underlying component (either `RCTUITextView` or `RCTUITextField`) and we need to apply the `showSoftInputOnFocus` prop again to the newly created view. Otherwise, it will be reset every time the `multiline` is set.